### PR TITLE
notifier: apply the filters in the same thread

### PR DESCRIPTION
**Pull request by anmic**

### CODE REVIEW

**Comment by anmic** [comment]
@vmihailenco PTAL

File: pybrake/notifier.py [hunk 1]
```diff
@@ -234,7 +243,12 @@ def send_notice(self, notice):
             f = futures.Future()
             f.set_result(notice)
             return f
-        return pool.submit(self.send_notice_sync, notice)
+
+        notice, ok = self._filter_notice(notice)
+        if not ok:
+            return pool.submit(lambda notice: notice, notice)
```
Comment (vmihailenco) [inline pybrake/notifier.py]: Ignored notice must NOT be sent to the pool.

**Comment by anmic** [comment]
Fixed. I don't send filtered notices to the pool.

**Comment by vmihailenco** [comment]
Create function `_send_notice_sync` which does exactly what `send_notice_sync` did but does not apply filter. Then use it in `send_notice` and `send_notice_sync`

**Pull request merged**